File: April-18/28-04-18/python/01-Play-with-numbers.py
n, q = map(int, input().split())
ar = [int(x) for x in input().split()]

ans = [0]*(n+1)

for i in range(1,n+1):
    ans[i] = ans[i-1] + ar[i-1]

for i in range(0,q):
    a,b = map(int, input().split())
    divi = b-a+1
    val = (ans[b]-ans[a-1])//divi
    print(val)

File: April-18/28-04-18/c++/01-Play-with-numbers.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    int n,q;
    scanf("%d",&n);
    scanf("%d",&q);
    int ar[n+1];
    for(int i = 1; i <= n; i++)
        scanf("%d",&ar[i]);
    
    double ans[n+1];
    ar[0] = 0;
    for(int i = 1; i <= n; i++)
    {
        ans[i] = ans[i-1] + ar[i];
    }
    
    while(q--)
    {
        int a,b;
        scanf("%d",&a);
        scanf("%d",&b);
        
        int divi = b-a+1;
        long val =  (ans[b] - ans[a-1]) / divi;
        printf("%ld\n",val);
    }
    
    
}
